WordPress/gutenberg
# Global Styles: Use low-level cache for get_user_data_from_wp_global_styles

Backports changes from https://github.com/WordPress/wordpress-develop/pull/3517

See also https://github.com/WordPress/gutenberg/issues/44946 and https://core.trac.wordpress.org/ticket/56901

## Proposed changes
This change removes caching of global styles for logged in users, allowing "wp_global_styles" custom post type to be imported completely, regardless of any previously cached data. This change now relies on the lower-level native WP_Query cache invalidation methods for the global styles post type.

## Testing Instructions

- Use Twenty Twenty Two (TT2) theme
- Use WordPress 6.0.3
- Log into the admin area
- Go to Appearance > Editor
- Open the Global Styles UI
- Select "Browse Styles"
- Select any non-default variation
- Select "Save" to save the new global styles
- Go back to the Dashboard
- Go to Tools > Export
- Click on "Download Export File" using "All content"
- Using another local site, with WordPress 6.1 RC3 (where the issue https://github.com/WordPress/gutenberg/issues/44946 is still present) and Gutenberg running this branch
- Log into the admin area
- Go to Tools > Import
- Select "Install Now" for the "WordPress" importer
- Select "Run Importer"
- Select the XML file using "Choose File"
- Click on the "Upload file and import" button
- View site in the frontend
- Make sure the styles use the non-default variation you selected
